### ISSUE TITLE
Issues with travis and coveralls

### ISSUE THREAD

**Issue by juanda99** [react-boilerplate/react-boilerplate#1477]
Trying to use travis and coveralls with this repo (v3.4).

If I fork the repo, and push it to GitHub in order to launch Travis, I got an error:

```
0.86s$ npm run coveralls
> react-boilerplate@3.4.0 coveralls /home/travis/build/juanda99/react-boilerplate
> cat ./coverage/lcov.info | coveralls
cat: ./coverage/lcov.info: No such file or directory
[error] "2017-01-17T20:42:33.504Z"  'error from lcovParse: ' 'Failed to parse string'
[error] "2017-01-17T20:42:33.506Z"  'input: ' ''
[error] "2017-01-17T20:42:33.507Z"  'error from convertLcovToCoveralls'
/home/travis/build/juanda99/react-boilerplate/node_modules/coveralls/bin/coveralls.js:18
        throw err;
        ^
Failed to parse string
```

I think lcov.info doesn't exist because npm script "test:clean" is called previous to coveralls so it deletes coverage directory:
`"build:clean": "npm run test:clean && rimraf ./build",`
I rewrite it as: 
`"build:clean": "rimraf ./build",`
And everything works ok.

However... If I decide to start my project and execute npm run clean, I have three issues ([see travis log](https://travis-ci.org/juanda99/arasaac-frontend/jobs/193205076)):
- Script ./internals/scripts/generate-templates-for-linting.js fails (it you create at least one component it doesn't fail anymore).
-   npm run tests also fails:
ERROR: Error: ENOENT: no such file or directory, scandir '/home/travis/build/juanda99/arasaac-frontend/node_modules/istanbul-reports/lib/html/assets'
If I execute npm run test in my machine there's no error.

-  npm run build also fails:
STACK: Error: ENOENT: no such file or directory, scandir '/home/travis/build/juanda99/arasaac-frontend/node_modules/istanbul-reports/lib/html/assets'
If I execute npm run build in my machine there's no error.

**Comment by juanda99** [comment]
I can send PR so build doesn't remove tests and coveralls process data. Just rewrite this line in package.json file:
"build:clean": "npm run test:clean && rimraf ./build",
I rewrite it as:
"build:clean": "rimraf ./build"

Some of the above errors (last two errors) where magically solved after cleaning the cache from travis.

However Just don't know why ./internals/scripts/generate-templates-for-linting.js is in .travis.yml, I can't see the benefits. It gives me some errors when doing CI:
- It fails after executing npm run clean (no components). As soon as I create a component it gets ok.
- If I change my eslint configuration, removing semicolons at the end of lines (rule  "semi": ["error", "never"],) it shows erros:

$ node ./internals/scripts/generate-templates-for-linting
/home/travis/build/CPIFP-Los-Enlaces/react-boilerplate/internals/generators/language/index.js:81
            throw err || stderr
            ^
Error: Command failed: npm run extract-intl
/home/travis/build/CPIFP-Los-Enlaces/react-boilerplate/app/i18n.js:48
  fr: formatTranslationMessages('fr', frTranslationMessages)
                                      ^
ReferenceError: frTranslationMessages is not defined
    at Object.<anonymous> (/home/travis/build/CPIFP-Los-Enlaces/react-boilerplate/app/i18n.js:35:39)
    at Module._compile (module.js:571:32)
    at loader (/home/travis/build/CPIFP-Los-Enlaces/react-boilerplate/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/home/travis/build/CPIFP-Los-Enlaces/react-boilerplate/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/travis/build/CPIFP-Los-Enlaces/react-boilerplate/internals/scripts/extract-intl.js:17:14)

**Comment by samit4me** [comment]
Hi @juanda99 thank you for reporting the issue. It looks like this started with [PR #1432](https://github.com/react-boilerplate/react-boilerplate/pull/1432/files) and has been resolved in [PR #1563](https://github.com/react-boilerplate/react-boilerplate/pull/1563/files).

This has been merged into `dev`, so closing this issue.

Great work on debugging the issue, I wish I spotted this earlier 👍

**Comment by gihrig** [comment]
@juanda99 

> However Just don't know why ./internals/scripts/generate-templates-for-linting.js is in .travis.yml, I can't see the benefits. It gives me some errors when doing CI:

More on this at #1223
